### ISSUE TITLE
Can RTMP connection be resumed after internet disconnection followed by reconnection?

### ISSUE THREAD

**Issue by avinashinde** [begeekmyfriend/yasea#426]
can RTMP connection can be resumed if there is wifi switch?

**Comment by begeekmyfriend** [comment]
In this demo you need to reconnect by hand. I think you can add such re-connection code in demo.

**Comment by shubhapp** [comment]
what you mean by reconnecting. should i start publishing again or just create to RTMP server ?

**Comment by begeekmyfriend** [comment]
I recommend restart the encoder as well because we still need SPS and PPS on reconnection.

**Comment by shubhapp** [comment]
yes i am doing the same. i am using following code to reconnect to stream 

` if (mFlvMuxer != null) {
            mFlvMuxer.start(url);
            startEncode();
        }`
But this block does not seems to work.

**Comment by begeekmyfriend** [comment]
Where is `mFlvMuxer.setVideoResolution`?

**Comment by shubhapp** [comment]
this I have already set when started publishing the first time. do I need to set it again?

**Comment by begeekmyfriend** [comment]
The encoder output resolution will be reset when the connection finishes.

**Comment by shubhapp** [comment]
Correct me if I am wrong for below steps for publishing video for RTMP 
 1. StartCamera
 2. Open socket Connection
 3. Complete Handshake -> 
 	a. send c0 
 	b. send c1 
 	c. receive s0
 	d. recieve s1
 	e. send c2
 	f. recieve s2
 4. start a loop to reac input packets 
 5 . create a stream
 	a. send release command
 	b. send PCPublish command
 	c. send create stream command (stream-id will be generated here )
 	d. wait for AMFO command (onStatus)

 6. start publishing packet after that 
 	a. sequence header Video/Audio packets 
 	b. video/audio packets 


So no if want reconnect with same video stream I cannot follow all above steps because it will create a different stream. But in case of reconnection is should use the same stream-id  

please guide what step I should follow in case of reconnection

**Comment by begeekmyfriend** [comment]
Basically you are right. The RTMP URL would not be changed on re-connection. The environment will be reset after `stopPublish` method to be invoked. Therefore what you need to do is just call these two methods as follows:
```java
startCamera // For preview
startPublish  // RTMP connection, starting encoding (to generate sequence header again)
```

**Comment by shubhapp** [comment]
If I do call stopPublish on internet disconnection then it will stop camera preview and onRtmpDisconnected() and onRtmpStopped will be called. But I do modify my publishing UI based on these events and there will be a no way knowing if the user has stopped in publishing or it is stopped because of internet disconnection. There should be difference how publishing has stopped

**Comment by begeekmyfriend** [comment]
You may notify UI in `handleException` in `MainAcitivity.java` which is part of demo for customization. In Yasea all exceptions thrown would serve based on Android message queue.
